docs/thick-plugin.md

```diff
@@ -53,38 +53,62 @@ kubectl apply -f deployments/multus-daemonset-thick.yml
-`-additional-bin-dir=/opt/multus/bin`, `-multus-log-level=debug`, etc.
-
-Furthermore, it also accepts a new command line parameter, where the user
-specifies the path to the server configuration:
+The available command line parameters are:
 
 - `config`: Defaults to `"/etc/cni/net.d/multus.d/daemon-config.json"`
-- `metricsPort`: Metrics port (of multus' metric exporter), default is disable
+- `version`: Prints the daemon config version and exits
 
-### Server configuration
+### Server / Daemon configuration
 
 The server configuration is encoded in JSON, and allows the following keys:
 
 - `"chrootDir"`: Specify the directory which points to host root from the pod. See 'Chroot configuration' section for the details.
-- `"socketDir"`: Specify the location where the unix domain socket used for
-client/server communication will be located. Defaults to `"/run/multus"`.
+- `"daemonSocketDir"`: Specify the location where the unix domain socket used
+for client/server communication will be located. This is the location where the
+**Daemon** will read the configuration from. Defaults to `"/run/multus"`.
+- `"metricsPort"`: Metrics port (of multus' metric exporter); by default, no port
+is provided.
+- `"logFile"`: the path to where the daemon logs will be persisted.
+- `"logLevel"`: the logging level for the multus daemon logs.
+- `"logToStderr"`: enable this to have the daemon multus logs echoed to stderr
+as well. By default, it is disabled.
 
 In addition, you can add any configuration which is in [configuration reference](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/configuration.md#multus-cni-configuration-reference). Server configuration override multus CNI configuration (e.g. `/etc/cni/net.d/00-multus.conf`)
 
+Below you can see an example of the daemon configuration:
+```json
+{
+        "chrootDir": "/hostroot",
+        "confDir": "/host/etc/cni/net.d",
+        "logToStderr": true,
+        "logLevel": "verbose",
+        "logFile": "/tmp/multus.log",
+        "binDir": "/opt/cni/bin",
+        "cniDir": "/var/lib/cni/multus",
+        "daemonSocketDir": "/host/run/multus/",
+        "cniVersion": "0.3.1",
+        "cniConfigDir": "/host/etc/cni/net.d",
+        "multusConfigFile": "auto",
+        "multusAutoconfigDir": "/host/etc/cni/net.d"
+    }
+```
+
+### Client / Shim configuration
+
+The multus shim configuration is encoded in JSON, and essentially is just a
+regular CNI configuration, usually available in `/etc/cni/net.d/00-multus.conf`.
+
+It allows the following keys:
+
+- `"cniVersion"`: the CNI version for the Multus CNI plugin.
+- `"logFile"`:  the path to where the daemon logs will be persisted.
+- `"logLevel"`: the logging level for the multus daemon logs.
+- `"logToStderr"`: enable this to have the daemon multus logs echoed to stderr
+  as well. By default, it is disabled.
+- `"shimSocketDir"`: Specify the location where the unix domain socket used for
```

Comment:
"shimSocketDir" is defined in ControllerNetConf , so "shimSocketDir" definition could go to daemon configuration

https://github.com/k8snetworkplumbingwg/multus-cni/blob/7cf1c2f0c2af0cdabb47e00d9a7fd8290981e197/pkg/types/types.go#L185-L204

Reply:
Ah! I see. 

You mean to say I should document the shim attribute in the daemon section ? 

I personally wouldn't do that, since it isn't actually **used** from the daemon - it's there just so the configurations between the shim / daemon align. I now realize that might not be needed: I think it's possible to actually remove it from [the code](https://github.com/k8snetworkplumbingwg/multus-cni/blob/6c2e09529bcfdf2f5b7b7999df6ee43080d8bd57/pkg/types/types.go#L201), since it is not being used. 

Thoughts @s1061123 ?

Reply:
Not everything. "logFile"/"logLevel" and others are actually in shim side config. I'm just curiouos that ShimSocketDir is in daemonConfig but you mentioned in shim config. I'm okey if you remove 'ShimSocketDir' code from daemon side and update document (remove 'ShimSocketDir').

Reply:
Will do! Thank you.

Just for clarification - the `ShimSocketDir` is still required on the shim side. We need to instruct multus CNI where the socket is found (and it is in a different path than the one exposed by the `daemonSocketDir`).

Reply:
I suppose we just name 'socketDir'. If you don't want to do that, we could name
- 'socketDir' in daemon config and 
- 'daemonSocketDir' in shim config

because from daemon config side, 'daemon socket' seems redundant (as usual httpd.conf uses 'Port', not 'httpPort').

Reply:
Makes sense; now that there aren't 2 socket dirs here, having the prefix stopped making sense.